apps/publisher/src/app.tsx

```diff
@@ -63,6 +63,10 @@ const MainSourceId = 'Main';
 const DisplayShareSourceId = 'DisplayShare';
 type PublisherState = SourceState;
 
+const date = new Date();
+const streamName =
+  typeof window !== 'undefined' ? import.meta.env.VITE_MILLICAST_STREAM_NAME || date.valueOf().toString() : undefined;
```

Comment:
Do we need to check `window` here?

Reply:
So, the reason I chose to check window was to run this code only once, without relying on `state` or `useEffect`. That being said, I am happy to not do it this way, if your advice is to stick to `state` and `effect`